script/sem-ver-action.py

```diff
@@ -0,0 +1,70 @@
+#  imports
+import os
+import git
+from git import Repo
+
+#  variables
+script_path = os.getcwd()
+remote_url = "git@github.com:shiri2001/training.git"
+branch = "main"  # noqa:E501 change later to main so it only runs on main
+repo = Repo(script_path)
+
+
+#  clone repository from github if needed
+def clone():
+    try:
+        Repo.clone_from(remote_url, script_path)
+    except git.exc.GitCommandError:
+        pass
+
+
+def main():
+    # logging bot into github using desired branch
```

Comment:
The main method should be as clean as possible. All code blocks related to business logic should be placed in dedicated functions or classes.

Reply:
is there a particular part that should be moved?

Reply:
@shiri2001 Excellent, now the main method is clean.
Let's tune this up.
Watch this -->
https://www.youtube.com/watch?v=pTB30aXS77U&ab_channel=ArjanCodes